io/io/src/TBufferJSON.cxx

```diff
@@ -570,6 +682,15 @@ TString TBufferJSON::ConvertToJSON(const void *obj, const TClass *cl, Int_t comp
       return TBufferJSON::ConvertToJSON(ptr, member, compact, arraylen);
    }
 
+   if (cl && cl->GetCollectionProxy() && dynamic_cast<TEmulatedCollectionProxy*>(cl->GetCollectionProxy())) {
```

Comment:
You can also use
```
cl->GetCollectionProxy()->GetProperties() & TVirtualCollectionProxy::kIsEmulated
```
Note that you also should consider having this protection only if you know (and/or can assume) that the parameter is indeed a compiled version of the container (for example because the typeinfo overload was called).

Reply:
> Note that you also should consider having this protection only if you know (and/or can assume)
> that the parameter is indeed a compiled version of the container (for example because the
> typeinfo overload was called).

At that place I want to create CollectionProxy - if it is not exists.
Are there other direct code for that?

Reply:
That depends.   A TClass for STL collection always has a collection proxy with it.   If you meant, once you discovered that the collection proxy is emulated (but you know that the collection exist in compiled code) and want to generate a compiled collection proxy then currently the only way is to call gInterpreter->GenerateDictionary which will create a shared library (the ability to generate the dictionary solely within the interpreter has been worked on but never finished).

Reply:
>  A TClass for STL collection always has a collection proxy with it. 

Is emulated collection proxy enough for streaming?
Somehow without this "magic" line I was not able to directly stream STL classes.

Reply:
It is enough to stream an 'emulated collection' (i.e. essentially always a std::vector in memory) but it will fail to stream a compiled collection (because it would use it as a std::vector).

So for the case you described, there is a compiled collection that is being streamer and for that you need a compiled collection proxy (i.e. a dictionary for the collection).